File: codeforces/93.add_odd_or_subtract_even_[1311A]/aoose.cpp
#include <iostream>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int t;
  unsigned int a, b;
  cin >> t;
  while (t--) {
    cin >> a >> b;
    if (a == b) {
      cout << 0 << endl;
    }
    else if (a > b) {
      if (((a % 2) == 0 && (b % 2) == 0) || ((a % 2) != 0 && (b % 2) != 0)) {
        cout << 1 << endl;
      }
      else {
        cout << 2 << endl;
      }
    }
    else {
      if (((a % 2) == 0 && (b % 2) == 0) || ((a % 2) != 0 && (b % 2) != 0)) {
        cout << 2 << endl;
      }
      else {
        cout << 1 << endl;
      }
    }
  }
  return 0;
}

File: codeforces/93.add_odd_or_subtract_even_[1311A]/aoose.py
# Author: José Rodolfo (jric2002)
t = int(input())
while (t):
  a, b = map(int, input().split(" "))
  if (a == b):
    print(0)
  elif (a > b):
    if (((a % 2) == 0 and (b % 2) == 0) or ((a % 2) != 0 and (b % 2) != 0)):
      print(1)
    else:
      print(2)
  else:
    if (((a % 2) == 0 and (b % 2) == 0) or ((a % 2) != 0 and (b % 2) != 0)):
      print(2)
    else:
      print(1)
  t -= 1